kernel/build-kernel.sh

```diff
@@ -0,0 +1,252 @@
+# Defult path to search patches to apply to kernel
+readonly default_patches_dir="${GOPATH}/src/${kernel_config_repo}/kernel/patches/"
+# Defult path to search config for kata
+readonly default_kernel_config_dir="${GOPATH}/src/${kernel_config_repo}/kernel/configs/"
+# 
+patches_path=""
+#
+hypervisor_target=""
+#
+arch_target=""
+#
+kernel_config_path=""
+# destdir
+DESTDIR="${DESTDIR:-/}"
+#PREFIX=
+PREFIX="${PREFIX:-/usr}"
+#project_name
+readonly project_name="kata-containers"
+
+
+
+source "${script_dir}/../scripts/lib.sh"
+
+usage()
+{
+cat <<EOT
+Overview:
+
+	Build a kernel for Kata Containers
+
+Usage:
+
+	$script_name [options] <command> <argument>
+
+Commands:
+
+- setup 
+
+- build
+
+- install
+
+Options:
+
+	-c <path>: Path to config file to build a the kernel
+	-h       : Display this help.
+	-k <path>: Path to kernel to build
+	-p <path>: Path to a directory with patches to apply to kernel.
+	-v       : Kernel version to use if kernel path not provided.
+
+Example:
+	
+
+EOT
+}
+
+check_version_yaml()
+{
+	info "check versions.yaml"
+	local versions_yaml="${GOPATH}/src/${runtime_repository}/versions.yaml"
+	if [ ! -f "$versions_yaml" ]; then
+		info "Download ${runtime_repository}"
+		go get "${runtime_repository}"
+	fi
+	[ -f "$versions_yaml" ] || die "failed to found ${versions_yaml}"
+	
+
+}
+
+get_kernel()
+{
+	local version="${1}"
+	#Remove extra 'v'
+	version=${version#v}
+
+	info "Download kernel version ${version}"
+
+	major_version=$(echo "${version}" | cut -d. -f1)
+	kernel_tarball="linux-${version}.tar.xz"
+
+	if [ ! -f "${kernel_tarball}" ];then
+		info "Download kernel"
+		curl -OL "https://www.kernel.org/pub/linux/kernel/v${major_version}.x/${kernel_tarball}"
+	else
+		info "kernel tarball already downloaded"
+	fi
+
+	curl -OL "https://cdn.kernel.org/pub/linux/kernel/v${major_version}.x/sha256sums.asc"
+	grep "${kernel_tarball}" sha256sums.asc > "${kernel_tarball}.sha256"
+
+	sha256sum -c "${kernel_tarball}.sha256"
+
+	tar xf ${kernel_tarball}
+
+	# update global variable 
+	kernel_path=$(cd "linux-${version}" && pwd)
+
+
+}
+
+get_default_kernel_config(){
+	local version="${1}"
+	#Remove extra 'v'
+	local version="${version#v}"
+
+	local hypervisor="$2"
+	local kernel_arch="$3"
+
+	[ -n "${version}" ] || die "kernel version not provided"
+	[ -n "${hypervisor}" ] || die "hypervisor not provided"
+	[ -n "${kernel_arch}" ] || die "kernel arch not provided"
+
+	major_version=$(echo "${version}" | cut -d. -f1)
+	minor_version=$(echo "${version}" | cut -d. -f2)
+	config="${default_kernel_config_dir}/${kernel_arch}_kata_${hypervisor}_${major_version}.${minor_version}.x"
+	[ -f "${config}" ] || die "failed to find default config ${config}"
+	echo "${config}"
+}
+
+get_config_version(){
+	config_version_file="${default_patches_dir}/../kata_config_version"
+	if [ -f "${config_version_file}" ];then
+		cat "${config_version_file}"
+	else
+		echo "unknown"
+	fi
+}
+
+main()
+{
+	while getopts "a:c:hk:t:p:v:" opt
```

Comment:
Nit: Not quite sorted alphabetically (`p:` + `t:` wrong way round :) - same comment for the `case` below.